storage/range_command.go

```diff
@@ -39,59 +39,103 @@ import (
 // executeCmd switches over the method and multiplexes to execute the
 // appropriate storage API command. It returns an error and, for some calls
 // such as inconsistent reads, the intents they skipped.
-func (r *Range) executeCmd(batch engine.Engine, ms *engine.MVCCStats, args proto.Request, reply proto.Response) ([]proto.Intent, error) {
+func (r *Range) executeCmd(batch engine.Engine, ms *engine.MVCCStats, args proto.Request) (proto.Response, []proto.Intent, error) {
+	// TODO(tamird): Remove this.
+	var reply proto.Response
```

Comment:
Does this work? Modifying a return value from a defer is one of those cases where named return values are necessary. I would keep the names on the return values but use `return nil, nil, err` instead of bare `return` (because bare `return` can have surprising results if `err` is shadowed)